src/style.rs

```diff
@@ -398,19 +538,28 @@ pub struct FlexboxLayout {
     ///
     /// 1.0 is the default value, and this value must be positive.
     pub flex_shrink: f32,
-    /// Sets the initial main axis size of the item
-    pub flex_basis: Dimension,
-    /// Sets the initial size of the item
-    // TODO: why does this exist as distinct from flex_basis? How do they interact?
-    pub size: Size<Dimension>,
-    /// Controls the minimum size of the item
-    pub min_size: Size<Dimension>,
-    /// Controls the maximum size of the item
-    pub max_size: Size<Dimension>,
-    /// Sets the preferred aspect ratio for the item
-    ///
-    /// The ratio is calculated as width divided by height.
-    pub aspect_ratio: Option<f32>,
+
+    // Grid container properies
+    /// Defines the track sizing functions (widths) of the grid rows
+    pub grid_template_rows: GridTrackVec<GridTrackSizingFunction>,
+    /// Defines the track sizing functions (heights) of the grid columns
+    pub grid_template_columns: GridTrackVec<GridTrackSizingFunction>,
+    /// Defines the size of implicitly created rows
+    pub grid_auto_rows: GridTrackVec<GridTrackSizingFunction>,
+    /// Defined the size of implicitly created columns
+    pub grid_auto_columns: GridTrackVec<GridTrackSizingFunction>,
+    /// Controls how items get placed into the grid for auto-placed items
+    pub grid_auto_flow: GridAutoFlow,
+
+    // Grid child properties
+    /// Defines which row in the grid the item should start at
+    pub grid_row_start: GridLine,
+    /// Defines which row in the grid the item should end at
+    pub grid_row_end: GridLine,
+    /// Defines which column in the grid the item should start at
+    pub grid_column_start: GridLine,
+    /// Defines which column in the grid the item should end at
+    pub grid_column_end: GridLine,
```

Comment:
Should this be split out into a separate type? 

Would it make sense to have a `FlexboxLayout` and a different `GridLayout`?

Reply:
If we really want to split it out, I imagine something like below would make sense. I worry that adding static typing constraints just makes thing unwieldy though. Especially as I imagine it will be rare for Taffy to be used directly by hand, and much more common for it to be translated from the surface syntax of another library/system. I wonder if it might not be better to be permission with what we accept, but print diagnostic information about ignored styles (or return it to the caller in a machine inspectable format).

```rust
enum InnerDisplay {
    Flex(FlexContainerStyle),
    Grid(GridContainerStyle),
    Leaf,
}

struct FlexContainerStyle {
    flex_direction: FlexDirection,
    flex_wrap: FlexWrap,

    align_content: AlignContent,
    align_items: AlignItems,
    justify_content: JustifyContent,
    justify_items: JustifyItems,
    gap: Size<Dimension>,
}


struct GridContainerStyle {
    grid_template_rows: GridTrackVec<GridTrackSizingFunction>,
    grid_template_columns: GridTrackVec<GridTrackSizingFunction>,
    grid_auto_rows: GridTrackVec<GridTrackSizingFunction>,
    grid_auto_columns: GridTrackVec<GridTrackSizingFunction>,
    grid_auto_flow: GridAutoFlow,

    align_content: AlignContent,
    align_items: AlignItems,
    justify_content: JustifyContent,
    justify_items: JustifyItems,
    gap: Size<Dimension>,
}

enum OuterDisplay {
    None,
    Root,
    FlexChild(FlexChildStyle),
    GridChild(GridChildStyle),
}

struct FlexChildStyle {
    flex_basis: Dimension,
    flex_grow: f32,
    flex_shrink: f32,
}

struct GridChildStyle {
    grid_row_start: GridLine,
    grid_row_end: GridLine,
    grid_column_start: GridLine,
    grid_column_end: GridLine,
}

struct CssStyle {
    /// What layout strategy should be used?
    outer_display: OuterDisplay,
    inner_display: InnerDisplay,

    // Position properties
    position_type: PositionType,
    position: Rect<Dimension>,

    // Size properies
    size: Size<Dimension>,
    min_size: Size<Dimension>,
    max_size: Size<Dimension>,
    aspect_ratio: Option<f32>,
    margin: Rect<Dimension>,
    padding: Rect<Dimension>,
    border: Rect<Dimension>,
}
```

Reply:
This code base is hard enough to maintain as it is; I'm alright to increase the static typing to make it harder for us to have bugs.

Like you say, most end users will be using taffy in a wrapped form.

Reply:
I think this design approach can make sense and in the direction I was picturing. 
It might be that we should wait until we have a 3rd layouting algorithm to make sure that what we build is general across the correct domain.

@alice-i-cecile any opinion?

Reply:
This is the direction I was picturing too. I'd like to avoid adding `InnerDisplay` and `OuterDisplay` for this PR; it's a lot of complexity and can be nicely severed.

Reply:
I now think we ought to do something similar to https://github.com/bevyengine/bevy/pull/5513, but I agree that it ought to be a separate PR (probably landing after this one, as it'll be easier to work out the correct splits for the styles once we have all of them).